scripts/add_electricity.py

```diff
@@ -801,6 +802,55 @@ def estimate_renewable_capacities_irena(
             ] * float(p_nom_max)
 
 
+def attach_enhanced_geothermal(n):
+
+    egs_potential = pd.read_csv(snakemake.input["egs_potentials"], index_col=[0, 1])
+    assert egs_potential.index.names == ['network_region', 'capital_cost[$/kW]']
+
+    idx = pd.IndexSlice
+
+    n.add(
+        "Bus",
+        "EGS",
+        carrier="geothermal heat",
+        unit="MWh_th",
+    )
+
+    n.add(
+        "Generator",
+        "EGS",
+        bus="EGS",
+        carrier="geothermal heat",
+        p_nom_extendable=True,
+    )
```

Comment:
I recognise that it's still work a progress, just sharing some feedback which cab be probably helpful for future improvements. Not sure I completely get the architectural idea of this fragment. Aren't generators implemented as links a few lines below?